src/eos-updater-poll-common.c

```diff
@@ -796,6 +899,24 @@ run_fetchers (EosMetadataFetchData *fetch_data,
   if (g_hash_table_size (source_to_update) > 0)
     {
       EosUpdateInfo *latest_update = NULL;
+      g_autofree gchar *booted_ref = NULL;
+      g_autoptr(GError) metrics_error = NULL;
+
+      /* Send metrics about our ref: this is the ref we’re going to upgrade to,
+       * which is usually the same as the one we’re currently on, but in
+       * the case where we are upgrading from a checkpoint, the ref that we
+       * are going to switch to. */
+      if (get_refspec_to_upgrade_on (NULL, NULL, &booted_ref, NULL, &metrics_error))
+        {
+          g_autoptr(EosMetricsInfo) metrics = NULL;
+
+          metrics = eos_metrics_info_new (booted_ref);
+          maybe_send_metric (metrics);
+        }
+      else
+        {
+          g_message ("Failed to get metrics: %s", metrics_error->message);
+        }
```

Comment:
This chunk looks like a dodgy merge: the metrics are reported by the `metrics_report_successful_poll() call immediately below.

Reply:
Yeah ... I did think that something was amiss when this came up in the merge conflicts whilst rebasing. Thanks for spotting.